ElectronClient/app/InteropServiceHelper.js

```diff
@@ -61,12 +61,12 @@ class InteropServiceHelper {
 							cleanup();
 						}
 					} else {
-						win.webContents.print(options, (success) => {
+						win.webContents.print(options, (success, reason) => {
 							// TODO: This is correct but broken in Electron 4. Need to upgrade to 5+
 							// It calls the callback right away with "false" even if the document hasn't be print yet.
 
 							cleanup();
-							if (!success) reject(new Error('Could not print'));
+							if (!success && reason !== 'cancelled') reject(new Error('Could not print', reason));
```

Comment:
Can the Error constructor take multiple parameters? I can't find confirmation for this on MDN. Otherwise, if `reason` is always a string, you could simply concatenate it to the message.

Reply:
The 'reason' is always a string, either "failed" or "cancelled" so we can append it.

'Error' only takes 'message', 'filename', 'linenumber'.

So when a print is cancelled, we should throw the error.  But should we display the warning dialog if the print is cancelled?

Reply:
Yes the logic is right, we indeed shouldn't show the dialog when it's cancelled. The only thing I wasn't sure is about the Error constructor parameters, but otherwise there's no issue.